src/Shared.CLI/Helpers/RepoSearch.cs

```diff
@@ -28,16 +25,16 @@ namespace Microsoft.CST.OpenSource.Shared
     /// </summary>
     public class RepoSearch
     {
-        public RepoSearch(IHttpClientFactory httpClientFactory)
-        {
-            HttpClientFactory = httpClientFactory;
-        }
-
-        public RepoSearch() : this(new DefaultHttpClientFactory())
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepoSearch"/> class.
+        /// </summary>
+        /// <param name="projectManagerFactory">The <see cref="ProjectManagerFactory"/> to generate the project managers with.</param>
+        public RepoSearch(ProjectManagerFactory? projectManagerFactory = null)
         {
+            _projectManagerFactory = projectManagerFactory ?? new ProjectManagerFactory();
         }
```

Comment:
Convention question regarding these helper / tool classes: how often do we call these constructors without providing a project manager value (i.e. this class is at the top of a dependency graph), vs. how often do these constructors get calls from other helper/tool classes which need to pass the same factory object reference (i.e. this class is not at the top of a dependency graph)?

I ask because I wonder whether we should just make the `ProjectManagerFactory` dependency non-nullable and required for all of the helper/tool classes. If we frequently re-use these classes from within other such classes, making this parameter required would allow the compiler to remind us that we need to pass a value in instead of accidentally not passing a value and using the default `ProjectManagerFactory` instance where we shouldn't.

Reply:
The helpers it should never be null, but I think it's ok for the tool classes to not take in a `ProjectManagerFactory` and it be null by default. We only would override those in unit test cases, no one consumes the CLI tools and would have to worry about it being the default factory.

Reply:
> I ask because I wonder whether we should just make the ProjectManagerFactory dependency non-nullable and required for all of the helper/tool classes.

For Helper classes I think this is fine.  The only references I see to creating this without project manager factory specified is in tests, where it probably especially makes sense to use a mock instead of defaults.

For the OSSGadget's themselves I do not agree.  I think more users than not will not make any configuration changes to the default manager set and we should keep the parameterless ctors.

Reply:
I made it non-nullable in RepoSearch and PackageDownloader.